django/db/backends/postgresql/base.py

```diff
@@ -179,6 +197,45 @@ class DatabaseWrapper(BaseDatabaseWrapper):
     ops_class = DatabaseOperations
     # PostgreSQL backend-specific attributes.
     _named_cursor_idx = 0
+    _connection_pools = {}
+
+    @property
+    def pool(self):
+        if not is_psycopg3:
+            return None
+
+        if self.alias not in self._connection_pools:
+            pool = None
+            pool_options = self.settings_dict.get("OPTIONS", {}).get("pool", None)
+
+            if pool_options is not None:
+                # Verify that we are not running with persistent connections
+                if self.settings_dict.get("CONN_MAX_AGE", 0) != 0:
+                    raise ImproperlyConfigured(
+                        "Pooling doesn't support persistent connections"
+                    )
+                if pool_options is True:  # simply sets the default options
+                    pool_options = {}
+
+                from psycopg_pool import ConnectionPool
```

Comment:
This import requires `psycopg[pool]` to be installed, I think it will be helpful to users to catch `ModuleNotFoundError` here and provide some helpful error explaining that `psycopg[pool]` needs to be installed. It would also be good to mention in the docs.

Reply:
Added to the docs - great catch thank you!
I can't find other places that we are catching a `ModuleNotFoundError` to add a more helpful message. So tempted not to (I would also need to add a test to cover it I think).

Reply:
>  I can't find other places that we are catching a ModuleNotFoundError to add a more helpful message. So tempted not to (I would also need to add a test to cover it I think).

You will get some hits if you search for `ImportError`, as `ModuleNotFoundError` is a newer more targeted subclass of `ImportError` and not often used yet. But a doc comment is good enough too.

Reply:
Updated